solutions/USACO jan19-grass-plantin.mdx

```diff
@@ -0,0 +1,102 @@
+---
+id: usaco-894
+source: USACO Silver 2019 January
+title: Grass Planting
+author: Aadit Ambadkar
+---
+
+## Solution 1 - Intended Solution
+
+$\mathcal{O}(N)$ using key insight.
+
+Let ```adj[i]``` represent the number of paths adjacent to node $i$ in the graph. Then, considering that node, and that node alone, you would need exactly ```adj[i]+1``` numbers to fill it. This is because you would need ```adj[i]``` for each of the adjacent nodes, and $1$ for the node itself. So, our answer is the maximum of all the adjacents, plus $1$. 
+
+```java
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.StringTokenizer;
+
+public class planting {
+    public static void main(String[] args) throws IOException {
+        BufferedReader f = new BufferedReader(new FileReader("planting.in"));
+        PrintWriter pw = new PrintWriter(new BufferedWriter(new FileWriter("planting.out")));
+        StringTokenizer st = new StringTokenizer(f.readLine());
+        int n = Integer.parseInt(st.nextToken());
+        if (n==1) {
+            pw.println(1);
+            pw.close();
+        } else if (n==2) {
+            pw.println(2);
+            pw.close();
+        }
+        int[] adj = new int[n];
+        Arrays.fill(adj, 1);
+        int max=1;
+        for (int i = 0; i < n-1; i++) {
+            st = new StringTokenizer(f.readLine());
+            int ne = Integer.parseInt(st.nextToken())-1;
+            if (++adj[ne]>adj[max]) max=ne;
+            ne = Integer.parseInt(st.nextToken())-1;
+            if (++adj[ne]>adj[max]) max=ne;
+        }
```

Comment:
why not an adj array but just keep track of the max value and add 1 at the end?

Reply:
I do think that is exactly what is happening here.

Reply:
use `ans = Math.max()` and do `ans+1` at the end.